### PR TITLE
Introduce ProtectedPropertyAccessor

### PR DESCRIPTION
**Pull request by gregpriday**
Make it possible to access the protected properties of `Struct` classes.

### CODE REVIEW

File: src/Models/Traits/StructProtectedPropertyAccessor.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        if (property_exists($this, $property)) {
+            $reflection = new ReflectionProperty($this, $property);
+            if ($reflection->isProtected()) {
+                return $this->$property;
+            } else {
+                throw new InvalidArgumentException("Access to property '$property' is not allowed");
+            }
+        }
+
+        throw new InvalidArgumentException("Property '$property' does not exist");
```
Comment (gregpriday) [inline src/Models/Traits/StructProtectedPropertyAccessor.php:33]: We could also potentially return `null` here.

**Comment by gregpriday** [comment]
@hkulekci Everything is tested and ready for review if you'd like to add this. Again, there are workarounds, but I'd personally find this very useful in my library.

**Comment by gregpriday** [comment]
I switched to using magic functions instead. Accessing protected properties, even though the trait would handle them, was causing IDE notices. These are a little annoying. Magic functions don't have the same issues.

**Comment by hkulekci** [comment]
Thanks. ✌️

**Pull request merged**